### PR TITLE
chore(typescript): transform components to typescript code standard

### PR DESCRIPTION
**Pull request by PeterMK85**
### Description

1-1 typescript transform of the components _(no logic change or similar)_

### Review

- [x] Verify that all classes affected by the changes have **good** top-level documentation.
- [ ] Verify that new code has been covered with specs (and features if applicable).
- [ ] Verify that a corresponding meaningful changelog entry has been added (`* Add Github pull request template`), for this change if applicable.
- [ ] Test manually.
- [ ] Get two :+1: from code review.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that all new dependencies are included in `package.json`.
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.
- [ ] Verify the branch name starts with `NNNN` where `NNNN` is the issue number.

### Screenshots

| Before                                        | After                                         |
| --------------------------------------------- | --------------------------------------------- |
| _Insert screenshots and/or screen recordings_ | _Insert screenshots and/or screen recordings_ |

### Other

_Provide additional notes, remarks, links, mention specific people to review,…_

### CODE REVIEW

File: src/calendar/header_button.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  arrow?: JSX.Element
+  blockClassName?: string
+  enabled?: boolean
+  offset?: number
+  onClick: (...args: any[]) => any
+  title?: string
+  type: 'prev' | 'next'
+}
+
+const HeaderButton: SFC<Props> = ({arrow, blockClassName, enabled, type, title, onClick}) => (
```
Comment (havenchyk) [inline src/calendar/header_button.tsx]:
```
 /**
     * @deprecated as of recent React versions, function components can no
     * longer be considered 'stateless'. Please use `FunctionComponent` instead.
     *
     * @see [React Hooks](https://reactjs.org/docs/hooks-intro.html)
     */
    type SFC<P = {}> = FunctionComponent<P>;
```

**Comment by havenchyk** [comment]
The only blocker here to me - proper types for components. Let me please go one by one to check them

**Comment by PeterMK85** [comment]
Sure @havenchyk, I do not want to waste time with binded functions, rest I went over, I consider it as a base.

I want to clean up the render props without any additional child etc, and move the logic into one container (rather than the current `calendar` and `month` split)

File: src/@types/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export type IDate = Date | number | string
+
+export type ISelectionRange = {
+  start: IDate
+  end: IDate
+  inProgress?: boolean
+}
+
+export enum IMode {
```
Comment (havenchyk) [inline src/@types/index.ts]: I would use `'range'|'single'` here, not enum

**Pull request merged**